app/assets/javascripts/polaris_view_components/autocomplete_controller.js

```diff
@@ -4,43 +4,49 @@ import { debounce } from './utils'
 
 export default class extends Controller {
   static targets = ['popover', 'input', 'results', 'option', 'emptyState']
-  static values = { url: String, selected: Array }
+  static values = { multiple: Boolean, url: String, selected: Array }
 
-  connect() {
-    this.inputTarget.addEventListener("input", this.onInputChange)
-  }
+  // Actions
 
-  disconnect() {
-    this.inputTarget.removeEventListener("input", this.onInputChange)
+  onFocus() {
+    this.showAutocomplete()
   }
 
-  // Actions
-
-  toggle() {
-    if (this.isRemote && this.visibleOptions.length == 0 && this.value.length == 0) {
+  onInput = debounce(() => {
+    if (!this.popoverController.isActive) {
```

Comment:
We only want to search if suggestion box is active as that's when we want to either fetch the data or filter the results.

If something else populates the input field, but the autocomplete suggestions are not open, we don't want to do anything here. This can happen when: user clicks on some kind of a "template" that prepopulates the data or...when user selects one of the autocomplete suggestions